src/walletConnect/screens/EstimatedNetworkFee.tsx

```diff
@@ -0,0 +1,84 @@
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import TokenDisplay from 'src/components/TokenDisplay'
+import { NETWORK_NAMES } from 'src/shared/conts'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { useTokenInfo } from 'src/tokens/hooks'
+import { getTokenId } from 'src/tokens/utils'
+import { NetworkId } from 'src/transactions/types'
+import { SerializableTransactionRequest } from 'src/viem/preparedTransactionSerialization'
+
+interface Props {
+  networkId: NetworkId
+  transaction: SerializableTransactionRequest
+}
+
+export default function EstimatedNetworkFee({ networkId, transaction }: Props) {
+  const { t } = useTranslation()
+
+  const feeTokenId =
+    'feeCurrency' in transaction
+      ? getTokenId(networkId, transaction.feeCurrency)
+      : getTokenId(networkId)
+
+  const feeTokenInfo = useTokenInfo(feeTokenId)
+
+  const networkName = NETWORK_NAMES[networkId]
+
+  if (!transaction.gas || !transaction.maxFeePerGas || !feeTokenInfo || !networkName) {
+    // insufficient information to display fee details
+    return null
```

Comment:
this is unexpected right? maybe we can log a warning or error?

Reply:
yes, normally this shouldn't happen. added a warning log.